reporter/html.js

```diff
@@ -610,6 +611,114 @@ function getNameHtml( name, module ) {
+	temp_array = [];
+	for(iter = 0; iter<str.length;iter++) {
+		split_array = [];
+		if(str[iter].indexOf("</del>")> -1) {
+			str[iter] = "<del>" + str[iter];
+			split_array = str[iter].split("</del>");
+			split_array[0] = split_array[0] + "</del>";
+			temp_array.push(split_array[0]);
+			temp_array.push(split_array[1]);
+		} else {
+			temp_array.push(str[iter]);
+		}
+	}
+	str = temp_array
+	temp_array = [];
+	split_array1 = [];
+	for(iter = 0; iter<str.length;iter++) {
+		split_array = [];
+		if(str[iter].indexOf("<ins>")> -1) {
+			split_array = str[iter].split("<ins>");
+			for(iter1 = 0; iter1<split_array.length;iter1++) {
+				split_array1 = [];
+				if(split_array[iter1].indexOf("</ins>")> -1) {
+					split_array[iter1] = "<ins>" + split_array[iter1];
+					split_array1 = split_array[iter1].split("</ins>");
+					split_array1[0] = split_array1[0] + "</ins>";
+					temp_array.push(split_array1[0]);
+					temp_array.push(split_array1[1]);
+				} else {
+					temp_array.push(split_array[iter1]);
+				}
+			}
+		} else {
+			temp_array.push(str[iter]);
+		}
+	}
+	str = temp_array;
+	temp_array = [];
+	for (iter = 0; iter<str.length;iter++ ) {
+		split_array=[];
+		if( (str[iter].indexOf("<del>")> -1) || (str[iter].indexOf("<ins>")> -1) ) {
+			temp_array.push(str[iter]);
+		} else {
+			if(str[iter].indexOf("\n")> -1) {
+				split_array = str[iter].split("\n");
+				for (iter1 = 0;iter1<split_array.length;iter1++) {
+					if ((split_array.length > 5) && (iter1 === split_array.length - 2)) {
+						split_array[iter1] = "</span>\n" + split_array[iter1];
+					} else {
+						split_array[iter1] = "\n" + split_array[iter1];
+					}
+				}
+				split_array1 = "";
+				for(iter1 = 0;iter1<split_array.length;iter1++) {
+					split_array1 += split_array[iter1];
+				}
+				temp_array.push(split_array1);
+			} else {
+				temp_array.push(str[iter]);
+			}
+		}
+	}
+	str = temp_array;
+	temp_array = [];
+	iter2;
+	split_array2=[];
+	for (iter = 0; iter<str.length;iter++ ) {
+		split_array=[];
+		if( (str[iter].indexOf("<del>")> -1) || (str[iter].indexOf("<ins>")> -1) ) {
+			temp_array.push(str[iter]);
+		} else {
+			if(str[iter].indexOf("</span>")> -1) {
+				split_array = str[iter].split("</span>");
+				for (iter1 = 0;iter1<split_array.length-1;iter1++) {
+					split_array[iter1] = split_array[iter1]+"</span>";
+					split_array2 = split_array[iter1].split("\n");
+					if (split_array2.length > 2) {
+						split_array[iter1] = "<span id='expander-"+(depthExpand)+"'>[+]</span>" +
+								" <span class='qunit-collapsed' id='expanding-" +(depthExpand++)+ "'>"
+								+ split_array[iter1];
+					}
+				}
+				split_array1 = "";
+				for(iter1 = 0;iter1<split_array.length;iter1++) {
+					split_array1 += split_array[iter1];
+				}
+				temp_array.push(split_array1);
+			} else {
+				temp_array.push(str[iter]);
+			}
+		}
+	}
+	str = temp_array;
+	diff = "";
+	for(iter = 0;iter<str.length;iter++) {
+		diff += str[iter];
+	}
+	return diff;
+}
```

Comment:
this is unpleasantly complex. I'll need more time to review the interactions on each loop.

There isn't a way to simplify this function? It would also be easier to review this if following the style guide.

Also, the very similar names like `split_array1` and `split_array` make this function harder to follow along.